### PR TITLE
Enable corepack & fix `pre-commit run --all-files`

### PR DESCRIPTION
**Pull request by Mikaela**
Briefly talked on Matrix, I think the project was previously using an old yarn version since there was a warning about the project being updated to new yarn version when I ran `corepack use yarn`.

I don't know if this brings breaking changes and I am curious on whether the tests or CI pass.

### CODE REVIEW

File: .yarnrc.yml [hunk 1]
```diff
@@ -0,0 +1 @@
+nodeLinker: node-modules
```
Comment (Mikaela) [inline .yarnrc.yml:1]: Specifying pnpm here might conserve disk space

- https://yarnpkg.com/configuration/yarnrc#nodeLinker

File: mx-tester.yml [hunk 2]
```diff
@@ -7,7 +7,7 @@ up:
     - until psql postgres://mjolnir-tester:mjolnir-test@127.0.0.1:8083/mjolnir-test-db -c ""; do echo "Waiting for psql..."; sleep 1s; done
     # Launch the reverse proxy, listening for connections *only* on the local host.
     - docker run --rm --network host --name mjolnir-test-reverse-proxy -p 127.0.0.1:8081:80 -v $MX_TEST_CWD/test/nginx.conf:/etc/nginx/nginx.conf:ro -d nginx
-    - yarn install
+    - corepack yarn install
     - npx ts-node src/appservice/cli.ts -r -u "http://host.docker.internal:9000"
```
Comment (Mikaela) [inline mx-tester.yml:11]: Isn't npx some npm command? Maybe this should also be `corepack yarn dlx`?

- https://yarnpkg.com/cli/dlx

**Comment by Mikaela** [comment]
Additionally it's not necessary to call `corepack` constantly since `corepack enable` should make `yarn` point to it, but I am choosing the lazy way out of having to figure out how to get the CI to use corepack version instead of Yarn Classic which is installed by default.

It's inspired by https://github.com/nodejs/corepack/blob/v0.28.2/README.md#corepack-enable--name

**Comment by Mikaela** [comment]
Scimming through the CI logs it seems that it's unable to find Draupnir version, which I think is related to this

- https://yarnpkg.com/migration/guide#explicitly-call-the-pre-and-post-scripts

I think that needs actual coder to check it thogh.

**Comment by Gnuxie** [comment]
Yeah that's a known issue and unrelated to this PR, don't worry about it

**Comment by Gnuxie** [comment]
@Mikaela did you follow https://yarnpkg.com/migration/guide ?

File: package.json [hunk 3]
```diff
@@ -77,5 +77,6 @@
   },
   "engines": {
     "node": ">=18.0.0"
-  }
+  },
+  "packageManager": "yarn@4.3.1+sha512.af78262d7d125afbfeed740602ace8c5e4405cd7f4735c08feb327286b2fdb2390fbca01589bfd1f50b1240548b74806767f5a063c94b67e431aabd0d86f7774"
```
Comment (Gnuxie) [inline package.json:81]: Why is there a sha512?

**Comment by Mikaela** [comment]
More or less, although by accident

File: package.json (hunk 3 above)
Comment (Mikaela) [reply package.json:81]: It is what corepack use yarn gave me

**Comment by Gnuxie** [comment]
oki

**Comment by Gnuxie** [comment]
Been playing around with yarn4 just to make sure everything is in order, unfortunately I haven't been able to get `yarn link` working with VSCode's debugger when linking MPS locally. And that seems to be because they no longer use a symlink in `node_modules`. 

https://classic.yarnpkg.com/lang/en/docs/cli/link/
https://yarnpkg.com/cli/link
https://yarnpkg.com/configuration/manifest#resolutions
https://yarnpkg.com/protocol/portal

I'm investigating this but it might take awhile.

**Comment by Mikaela** [comment]
https://yarnpkg.com/getting-started/editor-sdks ?

**Comment by Gnuxie** [comment]
Ok, that looks promising. However, seems like projects that have been touched by yarn classic aren't automatically migrated to use "yarn plug'n'play" which is required for the editor integration. https://yarnpkg.com/migration/pnp. 
And that migration guide requires some major work.

**Pull request closed without merging**